src/test/java/uk/gov/companieshouse/filinghistory/api/controller/FilingHistoryControllerMongoUnavailableIT.java

```diff
@@ -0,0 +1,180 @@
+    private static final String TRANSACTION_ID = "transactionId";
+    private static final String COMPANY_NUMBER = "12345678";
+    private static final String SELF_LINK = "/company/%s/filing-history/%s".formatted(COMPANY_NUMBER, TRANSACTION_ID);
+    private static final String ENTITY_ID = "1234567890";
+    private static final String DOCUMENT_ID = "000X4BI89B65846";
+    private static final String BARCODE = "X4BI89B6";
+    private static final String NEWEST_REQUEST_DELTA_AT = "20140916230459600643";
+    private static final String UPDATED_BY = "5419d856b6a59f32b7684d0e";
+    private static final String TM01_TYPE = "TM01";
+    private static final String DATE = "2014-09-15T23:21:18.000Z";
+    private static final String ORIGINAL_DESCRIPTION = "Appointment Terminated, Director john tester";
+    private static final String OFFICER_NAME = "John Tester";
+    private static final String RESIGNATION_DATE = "29/08/2014";
+    private static final String DESCRIPTION = "termination-director-company-with-name-termination-date";
+    private static final String ACTION_AND_TERMINATION_DATE = "2014-08-29T00:00:00.000Z";
+    private static final String CONTEXT_ID = "ABCD1234";
+    private static final String RESOURCE_CHANGED_URI = "/private/resource-changed";
+
+    @Container
+    private static final MongoDBContainer mongoDBContainer = new MongoDBContainer("mongo:5.0.12");
+
+    @Autowired
+    private MongoTemplate mongoTemplate;
+    @Autowired
+    private MockMvc mockMvc;
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @MockBean
+    private Supplier<Instant> instantSupplier;
+    @MockBean
+    private Repository repository;
+
+    @BeforeAll
+    static void start() {
+        System.setProperty("spring.data.mongodb.uri", mongoDBContainer.getReplicaSetUrl());
+    }
+
+    @BeforeEach
+    void setUp() {
+        mongoTemplate.dropCollection(FILING_HISTORY_COLLECTION);
+        mongoTemplate.createCollection(FILING_HISTORY_COLLECTION);
+    }
+
+    @Test
+    void shouldReturn503WhenRepositoryThrowsServiceUnavailableDuringSave() throws Exception {
+        // given
+        final InternalFilingHistoryApi request = buildPutRequestBody();
+
+        doThrow(ServiceUnavailableException.class)
+                .when(repository).save(any());
+
+        // when
+        ResultActions result = mockMvc.perform(put(PUT_REQUEST_URI, COMPANY_NUMBER, TRANSACTION_ID)
+                .header("ERIC-Identity", "123")
+                .header("ERIC-Identity-Type", "key")
+                .header("ERIC-Authorised-Key-Privileges", "internal-app")
+                .header("X-Request-Id", CONTEXT_ID)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(objectMapper.writeValueAsString(request)));
+
+        // then
+        result.andExpect(MockMvcResultMatchers.status().isServiceUnavailable());
+
+        verify(instantSupplier, times(1)).get();
```

Comment:
times(1) is the default so can be removed. Can also verify the repository mock is called with any().